kotlin/lib/src/main/kotlin/truelayer/signing/Exceptions.kt

```diff
@@ -0,0 +1,32 @@
+package truelayer.signing
+
+class InvalidKeyException private constructor(message: String, cause: Throwable) : Exception(message, cause) {
+    companion object {
+        internal fun <T> evaluate(block: () -> T): Result<T> = runCatching {
+            block()
+        }.mapThrowable { e -> InvalidKeyException("Invalid Key: ${e.message}", e) }
+    }
+}
+
+class InvalidSignatureException : Exception {
+    private constructor(message: String, cause: Throwable) : super(message, cause)
+    private constructor(message: String) : super(message)
+
+    companion object {
+        internal fun ensure(predicate: () -> Boolean, message: String): Result<Unit> =
+            if (predicate()) {
+                Result.success(Unit)
+            } else {
+                Result.failure(InvalidSignatureException("JWS signing/verification failed: $message"))
```

Comment:
I don't think we need the prefix, it already should be clear from the exception class
```suggestion
                Result.failure(InvalidSignatureException(message))
```